api/app/controllers/api/v1/positions_controller.rb

```diff
@@ -129,11 +145,35 @@ def matching_update_params
         end
 
         def positions_by_session
-            return Position.order(:id).select do |entry|
+            return all_positions.select do |entry|
                 entry[:session_id] == params[:session_id].to_i
             end
         end
 
+        def all_positions
+            exclusion = [:id, :created_at, :updated_at, :position_id]
+            return Position.order(:id).map do |entry|
+                matching = PositionDataForMatching.find_by(position_id: entry[:id])
+                matching = json(matching, except: exclusion)
+                ad = PositionDataForAd.find_by(position_id: entry[:id])
+                combined = json(ad, include: matching, except: exclusion)
+                combined = json(combined, include: {instructor_ids: entry.instructor_ids})
+                json(entry, include: combined, except: [:instructors])
+            end
+        end
+
+        def update_instructors_ids(position)
+            if params[:instructor_ids]
```

Comment:
you should use a guard clause for these kinds of nested conditionals, `return if params[:instructor_ids].blank?`

Reply:
Fixed.